File: 2 season/poo/database/camisa/java/draft.java
import java.util.Scanner;

public class draft {
    public static void main(String args[]) {
        Roupa camisa = new Roupa();
        Scanner input = new Scanner(System.in);

        while (camisa.getTamanho() == null) {
            System.out.println("Escolha um tamanho");
            camisa.setTamanho(input.nextLine());
        }

        input.close();
    }

    static class Roupa {
        private String tamanho;

        public Roupa() {
            this.tamanho = null;
        }

        public void setTamanho(String novo) {
            if (novo.equals("PP") || novo.equals("P") || novo.equals("M") 
                || novo.equals("G") || novo.equals("GG") || novo.equals("XG")) {
                this.tamanho = novo;
            } else {
                System.out.println("Valores permitidos: PP, P, M, G, GG, XG");
            }

        }

        public String getTamanho() {
            return this.tamanho;
        }
    }
}

File: 3 season/ed/ed/fuga/py/draft.py
h, p, f, d = map(int, input().split())

if d == 1:
    dh = (h - f) % 16
    dp = (p - f) % 16
elif d == -1:
    dh = (f - h) % 16
    dp = (f - p) % 16

if dp < dh:
    print("N")
else:
    print("S")